crates/astria-composer/src/searcher/executor/mod.rs

```diff
@@ -466,3 +467,30 @@ fn sha256(data: &[u8]) -> [u8; 32] {
     use sha2::Sha256;
     Sha256::digest(data)
 }
+
+/// A helper struct to report the number of sequence actions per rollup id in a bundle.
+struct ReportBundleRollupIdCounts<'a>(&'a [Action]);
+impl<'a> std::fmt::Display for ReportBundleRollupIdCounts<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use std::fmt::Write as _;
+
+        let mut counts = HashMap::new();
+        for action in self.0 {
+            if let Some(action) = action.as_sequence() {
+                *counts.entry(action.rollup_id).or_insert(0) += 1;
+            }
+        }
+
+        f.write_char('[')?;
+        let mut counts_iter = counts.iter();
+        if let Some((rollup_id, count)) = counts_iter.next() {
+            write!(f, "{rollup_id}: {count}")?;
+        };
+        for (rollup_id, count) in counts {
```

Comment:
in `counts` or `counts_iter`? does this double print the first/one of the rollups?

Reply:
Agree with @noot. This should consume `counts_iter`